go/api/config.go

```diff
@@ -0,0 +1,309 @@
+// Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0
+
+package api
+
+import "github.com/aws/glide-for-redis/go/glide/protobuf"
+
+const (
+	defaultHost = "localhost"
+	defaultPort = 6379
+)
+
+// NodeAddress represents the host address and port of a node in the cluster.
+type NodeAddress struct {
+	Host string // Optional: if not supplied, "localhost" will be used.
+	Port int    // Optional: if not supplied, 6379 will be used.
+}
+
+func (addr *NodeAddress) toProtobuf() *protobuf.NodeAddress {
+	if addr.Host == "" {
+		addr.Host = defaultHost
+	}
+
+	if addr.Port == 0 {
+		addr.Port = defaultPort
+	}
+
+	return &protobuf.NodeAddress{Host: addr.Host, Port: uint32(addr.Port)}
+}
+
+// RedisCredentials represents the credentials for connecting to a Redis server.
+type RedisCredentials struct {
+	// Optional: the username that will be used for authenticating connections to the Redis servers. If not supplied, "default"
```

Comment:
why document which fields are optional, when that is enforced by the constructor?

Reply:
Just for extra clarity, I can remove if you think its unnecessary.